### ISSUE TITLE
BrowserFrame with Resource in a Window issues a warning when window is closed

### ISSUE THREAD

**Issue by macAtNordea** [vaadin/framework#11369]
The issue is reproducible in Vaadin 7.7.15 and Vaadin 8.5.2

Reproduction steps:
1. Create a new Vaadin app with `vaadin-archetype-application`
2. Drop any pdf file to src/main/resources and rename it to `some.pdf`
3. Add commons-io as a dependency to your `pom.xml`
4. Use https://gist.github.com/macAtNordea/0330e9f06d17ec8d144b44cd28efa202 as the code for the UI depicting the issue. 
5. Run the app and click the button to open the window
6. Close the window

Observed results:
An additional request will be made for the PDF file displayed in the BrowserFrame that is embedded in the window. 
The server will answer with 404 as the resource will not be available anymore.
This is a problem for us. We have some  customizations, in case of 404 a default UI is shown to the user, but in this case the UI won't load. The bootstrap HTML will try to load ./VAADIN/vaadinBootstrap.js and fail:
![image](https://user-images.githubusercontent.com/45814343/49856947-8085d400-fdf1-11e8-9b9a-7925b6f1010d.png)

The reason can be found in the JS console:
![image](https://user-images.githubusercontent.com/45814343/49856990-a0b59300-fdf1-11e8-86c9-d686b6e1bbc3.png)

**Comment by macAtNordea** [comment]
I should add that this issue is observable at least on Google Chrome 71.0.3578.80, Windows 10

**Comment by TatuLund** [comment]
The problem is limited to the scenario when resource.setCacheTime(-1); is set. 

The root cause of the problem is that when iframe is being removed, it does another request, which results in error, since resource is not cached. I am not sure why and what is the purpose of it. However it seems to be Browser implementation and most of the browsers seem to have implemented this way.

Also it seems that when a Window been closed that wipes content rather forcefully. Hence for example beforeunload event is never fired from iframe's contentWindow, and it is hard to tackle this in BrowserFrame component itself by reseting url before iframe is unloaded. In theory that should be the fix for the issue, but as I tested it, it was not working.

But when Window is closed, a variable is sent to server and Window delegates Window removal to UI, namely removeWindow(..) method. This means, that we can write workaround by overriding removeWindow(..) in the UI of the application, see below:

    @Override
    public boolean removeWindow(Window window) {
    	frame.setSource(new ExternalResource("about:blank"));
    	push();
    	return super.removeWindow(window);
    }

So add this method to demo UI. Push needs to be enabled. Also, the forced push is needed in order to make sure that iframe is updated with blank content before actually wiping it away.

**Comment by macprzepiora** [comment]
I spent some time to research it and I found the following:

  * The content of the window is cloned so that it can be animated when the window is being actually closed and server-side components are already detached from the session. The cloning is being done in `postLayout()` method. The cloned node hierarchy is then attached to the DOM.
  * Unfortunately, reattaching an iframe causes it to reload itself, for details see for example https://bugzilla.mozilla.org/show_bug.cgi?id=254144. When the iframe cloned in `WindowConnector::cloneNodeFilteringMedia()` is attached it tries to load a resource that doesn't exist anymore on the server side.
* Setting `src` attribute of the cloned iframe to `about:blank` solves the issue. 

A patch for 7.x and 8.x will follow soon
